r-package/grf/vignettes/diagnostics.Rmd

```diff
@@ -31,22 +33,73 @@ Y <- pmax(X[, 1], 0) * W + X[, 2] + pmin(X[, 3], 0) + rnorm(n)
 cf <- causal_forest(X, Y, W)
 ```
 
-The overlap assumptions requires a positive probability of treatment for each $X_i$. We should not be able to deterministically decide the treatment status of an individual based on its covariates, meaning none of the estimated propensity scores should be close to one or zero. One can check this with a histogram:
+The overlap assumption requires a positive probability of treatment for each $X_i$. We should not be able to deterministically decide the treatment status of an individual based on its covariates, meaning none of the estimated propensity scores should be close to one or zero. One can check this with a histogram:
 
 ```{r}
 hist(e.hat <- cf$W.hat)
 ```
 
-One can also check that the covariates are balanced across the control and treated group by plotting the histogram $X_i | W_i = k$, overlaid here for each feature:
+One can also check that the covariates are balanced across the treated and control group by plotting the histograms of the inverse-propensity weighted samples $\frac{X_i | W_i = 1}{\hat e_i}$ and $\frac{X_i | W_i = 0}{1 - \hat e_i}$, overlaid here for each feature:
```

Comment:
We want a weighted histogram. E.g., for treated units, the value X[i] gets weight 1/e(X[i]) if treated (it's not that we make a histogram of X[i] / e(X[i])))